site/cds_rdm/tasks.py

```diff
@@ -140,3 +147,224 @@ def sync_users():
+        affiliations_to = name_to.get("affiliations", [])
+        for affiliation_from in name_from.get("affiliations", []):
+            if affiliation_from not in affiliations_to:
+                affiliations_to.append(affiliation_from)
+                updated = True
+
+        name_to["affiliations"] = affiliations_to
+
+        # Merge the identifiers
+        identifiers_to = name_to.get("identifiers", [])
+        for identifier_from in name_from.get("identifiers", []):
+            if identifier_from not in identifiers_to:
+                identifiers_to.append(identifier_from)
+                updated = True
+        name_to["identifiers"] = identifiers_to
+
+        # Merge the tags (except "unlisted" value)
+        tags_to = name_to.get("tags", [])
+        tags_from = name_from.get("tags", [])
+        for tag_from in tags_from:
+            if tag_from not in tags_to and tag_from != "unlisted":
+                tags_to.append(tag_from)
+                updated = True
+        name_to["tags"] = tags_to
+
+        # Merge the props
+        props_to = name_to.get("props", {})
+        for key, value in name_from.get("props", {}).items():
+            if key not in props_to:
+                props_to[key] = value
+                updated = True
+        name_to["props"] = props_to
+
+        # Mark as unlisted the name_from value
+        if "unlisted" not in tags_from:
+            if tags_from:
+                name_from["tags"] = tags_from + ["unlisted"]
+            else:
+                name_from["tags"] = ["unlisted"]
+
+            # "Deprecate" the name_from value
+            service.update(system_identity, name_from.get("id"), name_from)
+
+        if updated:
+            # Update the name_to value
+            service.update(system_identity, name_to.get("id"), name_to)
+
+    filters = [
+        dsl.Q("term", **{"props.is_cern": True}),
+        dsl.Q("term", **{"identifiers.scheme": "orcid"}),
+        dsl.Q("bool", must_not=[dsl.Q("term", tags="unlisted")]),
+    ]
+    if since:
+        filters.append(dsl.Q("range", updated={"gte": since}))
+    combined_filter = dsl.Q("bool", filter=filters)
+
+    names = service.scan(system_identity, extra_filter=combined_filter)
+
+    for name in names.hits:
+        orcid = next(
+            (
+                identifier.get("identifier")
+                for identifier in name.get("identifiers", [])
+                if identifier.get("scheme") == "orcid"
+            ),
+            None,
+        )
+        if orcid:
+            names_to_merge = service.resolve(
+                system_identity, id_=orcid, id_type="orcid", many=True
+            )
+            if names_to_merge.total > 1:
+                # We get the ORCID value and merge all the other values into it (Ideally there should be only 1 value apart from the ORCID)
+                orcid_name = None
+                other_names = []
+                for name_to_merge in names_to_merge.hits:
```

Comment:
sorry for a silly question, but I am not sure what condition we use to determine if we merge the name or not. Could we discuss this during the demo?

Reply:
I think that is not super clear just because there is a lot of code.
We merge when the CERN user has an ORCID in her/his profile, and the ORCID is found in the names vocab. from the ORCID dump. Basically, there is an ORCID match.

Reply:
maybe stating this in a comment in the code could help to understand whats the required condition to merge. I think it should be documented/destribed with more details